tests/test_hook_session_start_memory_redirect.py

```
@@ -0,0 +1,114 @@
+    assert legacy == "legacy memory"
+
+
+def test_fail_open_on_oserror(tmp_path: Path, monkeypatch) -> None:
+    """Write failures must not wedge the session (cross-platform)."""
+    home = tmp_path / "home"
+    home.mkdir()
+    monkeypatch.setenv("HOME", str(home))
+    repo = _setup_repo(tmp_path / "repo")
+    original_write = Path.write_text
+
+    def guarded_write(self: Path, data: str, *args, **kwargs) -> int:
```

Ruff (0.15.12)

[warning] Missing type annotation for `*args`

(ANN002)